leshan-core/src/main/java/org/eclipse/leshan/core/request/execute/Arguments.java

```diff
@@ -0,0 +1,105 @@
+package org.eclipse.leshan.core.request.execute;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class Arguments implements Iterable<SingleArgument> {
```

Comment:
First time, I see it. I'm asking if we really need an `Arguments` class and a `SingleArgument` one but maybe it's OK.

Reply:
At first glance Arguments are just list of arguments. But java list have many methods that should not be allowed, for example add() - we don't want to extend argument list once created. We could override add() method with UnsupportedOperationException, but the better is to disable this operation at compile time than runtime.